application/src/main/java/org/thingsboard/server/actors/ruleChain/DefaultTbContext.java

```diff
@@ -159,7 +162,7 @@ private void tellNext(TbMsg msg, Set<String> relationTypes, Throwable th) {
     @Override
     public void tellSelf(TbMsg msg, long delayMs) {
         //TODO: add persistence layer
```

Comment:
Remove the //TODO

Reply:
Maybe it's better to replace it with new todo: //TODO: fix all nodes that use this method to use ruleNodeCacheService for message persistence. See the example in the TbAbstractCacheBasedRuleNode.